File: gitdot-web/packages/gitdot-api/src/endpoint/repository/get-repository-commit-blobs.ts
import { z } from "zod";
import { RepositoryBlobPairResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetRepositoryCommitBlobsRequest = z.object({});
export type GetRepositoryCommitBlobsRequest = z.infer<
  typeof GetRepositoryCommitBlobsRequest
>;

export const GetRepositoryCommitBlobsResponse = z.array(
  RepositoryBlobPairResource,
);
export type GetRepositoryCommitBlobsResponse = z.infer<
  typeof GetRepositoryCommitBlobsResponse
>;

export const GetRepositoryCommitBlobs = {
  path: "/repository/{owner}/{repo}/commits/{sha}/blobs",
  method: "GET",
  request: GetRepositoryCommitBlobsRequest,
  response: GetRepositoryCommitBlobsResponse,
} as const satisfies Endpoint;
export type GetRepositoryCommitBlobs = typeof GetRepositoryCommitBlobs;

File: gitdot-api/src/endpoint/repository/get_repository_commit_blobs.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::repository::RepositoryBlobPairResource};

pub struct GetRepositoryCommitBlobs;

impl Endpoint for GetRepositoryCommitBlobs {
    const PATH: &'static str = "/repository/{owner}/{repo}/commits/{sha}/blobs";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetRepositoryCommitBlobsRequest;
    type Response = GetRepositoryCommitBlobsResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetRepositoryCommitBlobsRequest {}

pub type GetRepositoryCommitBlobsResponse = Vec<RepositoryBlobPairResource>;
